### PR TITLE
Add appuser support to aci-go-client

### PR DESCRIPTION
**Pull request by gshenoy**

### CODE REVIEW

File: client/auth.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ func (client *Client) InjectAuthenticationHeader(req *http.Request, path string)
 			Value: client.AuthToken.Token,
 		})
 		return req, nil
-	} else if client.privatekey != "" && client.adminCert != "" {
+	} else if client.privatekey != "" && (client.adminCert != "" || client.appUserName != "") {
```
Comment (tycha-d-1-3) [inline client/auth.go:63]: this doesn't look correct.  I know you're trying to pass in the appUserName twice, but why not pass i nthe adminCert name instead.  If you pass in the adminCertName, then you dont need this change here.

File: client/auth.go [hunk 2]
```diff
@@ -96,10 +96,17 @@ func (client *Client) InjectAuthenticationHeader(req *http.Request, path string)
 			Name:  "APIC-Certificate-Fingerprint",
 			Value: "fingerprint",
 		})
-		req.AddCookie(&http.Cookie{
+		if client.appUserName != "" {
+		   req.AddCookie(&http.Cookie{
+			Name:  "APIC-Certificate-DN",
+			Value: fmt.Sprintf("uni/userext/appuser-%s/usercert-%s", client.appUserName, client.appUserName),
```
Comment (tycha-d-1-3) [inline client/auth.go:102]: With the first comment, if you pass in both appUserName and adminCert, the only different would be the dn for the appuserName.  That would make this more flexible when you actually need to pass in different usercert names for the appUser.

**Review by tycha-d-1-3 — CHANGES_REQUESTED** [review]

File: client/auth.go (hunk 1 above)
Comment (gshenoy) [reply client/auth.go]: Removed appusername here.

File: client/auth.go (hunk 2 above)
Comment (gshenoy) [reply client/auth.go]: agreed.

**Review by tycha-d-1-3 — APPROVED** [review]

**Pull request merged**